code/datums/ai/sanity/_sanityloss_controller.dm

```diff
@@ -155,48 +177,138 @@
 	if(get_dist(living_pawn, current_movement_target) > max_target_distance)
 		CancelActions()
 		pathing_attempts = 0
-	if(our_turf == get_turf(living_pawn) && !isliving(current_movement_target))
+		return
+	if(our_turf == get_turf(living_pawn) && !isliving(current_movement_target) && !ismecha(current_movement_target))
 		if(++pathing_attempts >= MAX_PATHING_ATTEMPTS)
 			CancelActions()
 			pathing_attempts = 0
-
+			return
 	return TRUE
 
+/turf/proc/reachableTurftestWithMobs(caller, turf/T, ID, simulated_only)
+	if(T && !T.density && !(simulated_only && SSpathfinder.space_type_cache[T.type]) && !LinkBlockedWithAccess(T,caller, ID))
+		for(var/mob/living/L in T)
+			if(L.density)
+				return FALSE
```

Comment:
More appropriate would be checking CanPass proc here from what I recall.

Reply:
yea thats much better